Assets/Scripts/SS3D/Systems/Substances/SubstanceContainerActor.cs

```diff
@@ -0,0 +1,618 @@
+        {
+            substanceContainer.RemoveSubstance(substance, moles);
+        }
+
+        /// <summary>
+        /// Empties the container
+        /// </summary>
+        public void Empty()
+        {
+            substanceContainer.Empty();
+        }
+
+        /// <summary>
+        /// Removes moles from the container
+        /// </summary>
+        /// <param name="moles">The amount of moles</param>
+        [Server]
+        public void RemoveMoles(float moles)
+        {
+            substanceContainer.RemoveMoles(moles);
+        }
+
+        /// <summary>
+        /// Transfers moles to a different container
+        /// </summary>
+        /// <param name="other">The other container</param>
+        /// <param name="moles">The moles to transfer</param>
+        [Server]
+        public void TransferMoles(SubstanceContainerActor other, float moles)
+        {
+            substanceContainer.TransferMoles(other.substanceContainer, moles);
+        }
+
+        /// <summary>
+        /// Transfers volume (ml) to a different container
+        /// </summary>
+        /// <param name="other">The other container</param>
+        /// <param name="milliliters">How many milliliters to transfer</param>
+        [Server]
+        public void TransferVolume(SubstanceContainerActor other, float milliliters)
+        {
+            substanceContainer.TransferVolume(other.substanceContainer, milliliters);
+        }
+
+        /// <summary>
+        /// Returns the index of an substance
+        /// </summary>
+        /// <param name="substance">The substance to look for</param>
+        /// <returns>The index or -1 if it is not found</returns>
+        [Server]
+        public int IndexOfSubstance(Substance substance)
+        {
+            return substanceContainer.IndexOfSubstance(substance);
+        }
+
+        /// <summary>
+        /// Informs the system that the contents of this container have changed
+        /// </summary>
+        [Server]
+        public void MarkDirty()
+        {
+            OnContentsChanged();
+        }
+
+        [Server]
+        protected virtual void OnContentsChanged()
+        {
+             ProcessContainer(this);
+            ContentsChanged?.Invoke(this);
+        }
+
+        public override IInteraction[] CreateTargetInteractions(InteractionEvent interactionEvent)
+        {
+            return new IInteraction[]
+            {
+                new TransferSubstanceInteraction()
+            };
+        }
+
+        /// <summary>
+        /// Processes substances in a container
+        /// </summary>
+        public static void ProcessContainer(SubstanceContainerActor container)
+        {
+            SubstanceContainer.ProcessContainer(container.substanceContainer);
+            container.MarkDirty();
+        }
+
+        #endregion
+
+        #region SubstanceContainer
+        public class SubstanceContainer
```

Comment:
Move to another file.

Reply:
read what I wrote above

Reply:
my point stands.

Reply:
@joaoburatto except that's not a point, that's just telling to do something. I can't "just move" the file. If I do I have to break either : 
- encapsulation
- testability

I suggested an alternative solution to the nested class which would be public internal variable. So there's a whole discussion here. I'd be happy if you can find a better solution, I don't have anything except those two alternatives.

Reply:
Yeah well that's outdated. And still no point in sight.

Reply:
@joaoburatto Please respond to Nath about this in discord while he's online so we can resolve this. He would like an explanation greater than "my point still stands" without following up on your point at all.

As the code lead, one of your responsibilities is to see that PRs (at least technical ones) are getting processed timely. Maintainers will absorb much of that work but you also need to oversee them at times. 

Tbf you overall do a good job but this PR is starting to smell it's so old. Let's wrap it up!

Reply:
(You reviewed it right when I commented lol)